base_layer/core/src/transactions/transaction_components/transaction_kernel.rs

```diff
@@ -178,22 +181,43 @@ impl TransactionKernel {
     ///  Lock height
     ///  Features of the kernel
     ///  Burn commitment if present
-    pub fn build_kernel_challenge(
+    pub fn build_kernel_signature_challenge(
+        version: &TransactionKernelVersion,
         sum_public_nonces: &PublicKey,
         total_excess: &PublicKey,
         fee: MicroTari,
         lock_height: u64,
         features: &KernelFeatures,
         burn_commitment: &Option<Commitment>,
     ) -> [u8; 32] {
-        DomainSeparatedConsensusHasher::<TransactionHashDomain>::new("kernel_signature")
+        let message =
+            TransactionKernel::build_kernel_signature_message(version, fee, lock_height, features, burn_commitment);
+        let common = DomainSeparatedConsensusHasher::<TransactionHashDomain>::new("kernel_signature")
+            .chain(&version)
             .chain(sum_public_nonces)
             .chain(total_excess)
+            .chain(&message);
+        match version {
+            TransactionKernelVersion::V0 => common.finalize(),
+        }
+    }
+
+    /// Helper function to create the message of the challenge
+    pub fn build_kernel_signature_message(
```

Comment:
Can you document why these particular fields are included in this hash, but not others?